pkg/load/configAgent.go

```diff
@@ -145,16 +157,19 @@ func reloadWatcher(ctx context.Context, w *fsnotify.Watcher, c coalescer.Coalesc
 			if event.Op == fsnotify.Remove {
 				log.Debugf("Removing %s from watches", event.Name)
 				if err := w.Remove(event.Name); err != nil {
+					a.recordError("failed to remove item from watcher")
 					log.WithError(err).Errorf("Failed to remove %s from watches", event.Name)
 				}
 			}
 			go c.Run()
 			// add new files to be watched; if a watch already exists on a file, the
 			// watch is simply updated
 			if err := populateWatcher(w, path); err != nil {
+				a.recordError("failed to update watcher")
 				log.WithError(err).Error("Failed to update fsnotify watchlist")
 			}
 		case err := <-w.Errors:
+			a.recordError(fmt.Sprintf("received fsnotify error: %v", err))
```

Comment:
We should avoid label values that change -- it's easy to overdo the cardinality in Prometheus and crash it.